### PR TITLE
[SWA-185][FEAT] - Strategies module + "Popular Strategies" tab

### PR DESCRIPTION
**Pull request by ElRodrigote**
## Fixes: [SWA-185](https://linear.app/swaprhq/issue/STK-185/add-popular-strategies-strategy-tab)

# Description
* Add Sparkles icon ✨ .
* Abstract token information into constants.
* Abstract token logo sizes into a single source of truth (a single type).
* Minor modifications in `CaptionText` and `Button` components for compatibility.
* Create strategies constants, the 5th one is for test purposes. ⚠️ SHOULD BE REMOVED BEFORE MERGING ⚠️ 
* Create strategies context to handle changes across many modules.
* Create `StrategyCard` component and `Strategies` module.
* Wire up new `Strategies` module behavior into the `Stackbox` component.

# Visual evidence
https://www.loom.com/share/463abf3f963444239ad4c11717a129c7?sid=7c2766e9-b133-497f-9039-f8418a52c1bc

# How to test the changes

1) Pull this branch
2) Run the project locally
3) Go to Swapr landing page
4) Connect your wallet
5) Play around selecting and deselecting strategy cards, try updating any of the inputs from the stackbox.

### CODE REVIEW

File: packages/app/components/tabs/constants.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { twMerge } from "tailwind-merge";
+
+import { buttonStyles } from "@/ui";
+
+export const tabButtonStyles = twMerge(
+  buttonStyles({ variant: "secondary" }),
+  "bg-transperant border border-transparent text-em-high font-semibold ui-selected:bg-surface-75 ui-not-selected:text-em-low  hover:bg-surface-50 hover:border-surface-75 focus:ring-0"
+);
```
Comment (Diogomartf) [inline packages/app/components/tabs/constants.ts:8]: this doesn't make sense to be a constant. It's a style.

Renaming it for styles.ts would be an improvement.

File: packages/app/contexts/StrategyContext.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  sellToken: Token;
+  totalSellAmount: string;
+};
+
+const throwStrategyContextError = () => {
+  throw new Error("No StrategyContext available");
+};
+
+interface StrategyContextProps {
+  selectedStrategy: Strategy | null;
```
Comment (Diogomartf) [inline packages/app/contexts/StrategyContext.tsx:22]:
```suggestion
  selectedStrategy?: Strategy;
```

File: packages/app/contexts/StrategyContext.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  setSelectedStrategy: React.Dispatch<React.SetStateAction<Strategy | null>>;
+  setShouldResetStackbox: React.Dispatch<React.SetStateAction<Boolean>>;
+  shouldResetStackbox: Boolean;
+}
+
+const StrategyContext = createContext<StrategyContextProps>({
+  selectedStrategy: null,
```
Comment (Diogomartf) [inline packages/app/contexts/StrategyContext.tsx:29]:
```suggestion
```

**Review by berteotti — DISMISSED** [review]
Apart from Diogo style comment, LGTM

File: packages/app/components/tabs/constants.ts (hunk 1 above)
Comment (ElRodrigote) [reply packages/app/components/tabs/constants.ts]: Done!

File: packages/app/contexts/StrategyContext.tsx (hunk 2 above)
Comment (ElRodrigote) [reply packages/app/contexts/StrategyContext.tsx:22]: After an off-line talk we decided to keep this as is.

File: packages/app/contexts/StrategyContext.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  setSelectedStrategy: throwStrategyContextError,
+  setShouldResetStackbox: throwStrategyContextError,
+  shouldResetStackbox: false,
+});
```
[the remark this answers is not in the document]
Comment (ElRodrigote) [reply packages/app/contexts/StrategyContext.tsx:33]: After an off-line talk we decided to keep this as is.

**Review by Diogomartf — APPROVED** [review]

**Pull request merged**